lib/flagger/cloud.rb

```diff
@@ -246,41 +287,141 @@ def identify_entity(entity_json)
       end
     end
 
-    {
-      'treatment' => 'duration__get_treatment',
-      'payload' => 'duration__get_payload',
-      'eligible?' => 'duration__is_eligible',
-      'enabled?' => 'duration__is_enabled',
-    }.each do |name, stat_name|
-      define_method(name.to_sym) do |flagger_flag, entity_json|
-        stat = FlaggerUtils::Stat.new(stat_name)
-        stat.start()
-
-        entity = identify_entity(entity_json)
-        flag = @gating_info&.flag(flagger_flag.flag_name)
-        if flag.nil? && !@ingested_flags.include?(flagger_flag.flag_name) then
-          save_flag(flagger_flag.flag_name)
-        end
-        return_value = flag&.public_send(name, entity)
-
-        if name == 'treatment' and !return_value.nil? then
-          exposure = {
-            type: entity.type,
-            id: entity.id,
-            treatment_id: return_value.treatment_id,
-            treatment: return_value.codename
-          }
+    def treatment(flagger_flag, entity_json)
+      stat = FlaggerUtils::Stat.new('duration__get_treatment')
+      stat.start()
 
-          return_value = return_value.codename
+      entity = identify_entity(entity_json)
+      flag = @gating_info&.flag(flagger_flag.flag_name)
+      if flag.nil? && !@ingested_flags.include?(flagger_flag.flag_name) then
+        save_flag(flagger_flag.flag_name)
+      end
+      return_value = flag&.public_send('treatment', entity)
```

Comment:
We don't need to use `public_send` if the method name is static.
```suggestion
      return_value = flag&.treatment(entity)
```